src/lld/moviebooking/class_diagram.puml

```diff
@@ -1,3 +1,323 @@
 @startuml
+'Main Entity and Relational mapping entity
 
+package com.movie_booking.entity.main{
+    class Identifier{
+        - id: long
+        - name: String
+    }
+    class User extends Identifier{
+        - email: String
+        - phoneNo: String
+    }
+    class Movie extends Identifier{
+        - crew: List<Crew>
+        - supportedLanguages: List<LanguageCode>
+    }
+    class Crew extends Identifier{
+        - crewType: CrewType
+    }
+    enum CrewType{
+        Cameraman
+        Designer
+        Hero
+        Heroine
+        Director
+        Singer
+    }
+    'Todo: Preferred hierarchical approach, should discuss and final best entity design for city state, this takes four level joins
+    class District extends Identifier{
+        cityId: long
+    }
+    class City extends Identifier{
+        stateId: long
+    }
+    class State extends Identifier{
+        countryId: Country
+    }
+    class Country extends Identifier{
+    }
+
+    class Theatre extends Identifier {
+        - theatreGroupId: long
+        - address: Address
+        - screens: List<Screen>
+        - theatreStatus: TheatreStatus
+    }
+    class TheatreGroup extends Identifier{
+        - founder: String
+        - foundedDate: LocalDateTime
+    }
+    class Address{
+        - id: long
+        - addressLine1: String
+        - addressLine2: String
+        - districtId: districtId
+        - gpsLocation: GpsLocation
+        - pinCode: String
+    }
+    class Screen extends Identifier{
+        - seats: List<Seat>
+        - screenStatus: ScreenStatus
+    }
+    enum TheatreStatus{
+        Available
+        NotAvailable
+    }
+    enum ScreenStatus{
+        Available
+        NotAvailable
+    }
+    'Todo: Seat arrangement belongs to the screen style, user can customise it'
+    class Seat extends Identifier{
+        - rowNo: String
+        - colNo: String
+        - seatStatus: SeatStatus
+    }
+    enum SeatStatus{
+        Available,
+        NotAvailable,
```

Comment:
there needs to be clear distinction in naming between in progress booking, in this case, locked and not available(in this case booked)

if not availabele due to other reasons, it is better to add them as separate enum value